pkg/daemon/update.go

```diff
@@ -1280,9 +1280,25 @@ func (dn *Daemon) updateOS(config *mcfgv1.MachineConfig) error {
 	}
 
 	glog.Infof("Updating OS to %s", newURL)
-	if err := dn.NodeUpdaterClient.RunPivot(newURL); err != nil {
-		MCDPivotErr.WithLabelValues(newURL, err.Error()).SetToCurrentTime()
-		return fmt.Errorf("failed to run pivot: %v", err)
+	// In the cluster case, for now we run indirectly via machine-config-daemon-host.service
+	// for SELinux reasons, see https://bugzilla.redhat.com/show_bug.cgi?id=1839065
+	if dn.kubeClient != nil {
```

Comment:
does it make sense to instead check for /etc/ignition-machine-config-encapsulated.json file existance?

Reply:
Should be an equivalent condition - `kubeClient` is only set when we're running as the MCD and are connected to the cluster.  

Though I think `kubeClient` is *also* unset in the "once from" mode used for Ansible joining rhel7 workers, but for rhel7 workers we won't try running rpm-ostree either.

But...we are definitely inconsistent about these checks; some look for `dn.kubeClient` some look for `dn.recorder` etc.  It might be better to have a single `dn.runningAsPod` or something?

Reply:
Anything is fine by me as far as it works for all conditions. `ignition-machine-config-encapsulated.json` is just easier for me to understand/debug when something goes wrong :laughing: